python/cuml/tests/test_linear_model.py

```diff
@@ -561,6 +561,9 @@ def test_logistic_regression(
     )
     assert len(np.unique(cu_preds)) == len(np.unique(y_test))
 
+    if fit_intercept is False:
+        assert np.array_equal(culog.intercept_, sklog.intercept_)
```

Comment:
I thought the point of this change was to ensure that `intercept_` is equal in both cases. Why are we only testing for equality when `fit_intercept is False` ?

Reply:
When `fit_intercept is True`, `intercept_` can be a list of non-zero. It is possible that culog.intercept_ and sklog.intercept_ are different, given their implementations are not exactly the same. 

When 'fit_intercept is False', 'intercept_' will be a list of zero. The test case is able to assert equal the culog.intercept_ and sklog.intercept_.

Reply:
But they would still be similar, no? You can use the `cuml.testing.utils.array_equal` function like [here](https://github.com/rapidsai/cuml/blob/18acacf6247c211c97930864b2f4b1b3e076f09f/python/cuml/tests/test_linear_model.py#L192).

Reply:
Had tried assert array_equal(culog.intercept_, sklog.intercept_, 1e-3, with_sign=True).
Some tests failed: 
`FAILED tests/test_linear_model.py::test_logistic_regression[column_info0-1000-2-float32-none-1.0-True-1.0-0.001] - AssertionError: assert <array_equal: [0.60477465] [0.6086404] unit_tol=0.001 total_tol=0.0001 with_sign=True>`

`FAILED tests/test_linear_model.py::test_logistic_regression[column_info0-1000-10-float32-l1-1.0-True-1.0-0.001] - AssertionError: assert <array_equal: [-0.15404558  0.10340142  0.12519604 ... -0.13840534  0.13014889 -0.01927175] [-0.14051172  0.11931933  0.14109698 ... -0.12780005  0.14333196 -0.0074189 ] unit_tol=0.001 t...`

Reply:
Mean accuracies of culog and sklog are the same though. It seems the models converged to different optima.

Reply:
Ok, that's a valid point.